src/components/backoffice/parametres/ParametresLayout/ParametresHelpCard/ParametresHelpModal/ParametresHelpModal.tsx

```diff
@@ -1,28 +1,41 @@
-import React from 'react';
-import { useProfile } from '../../../useUpdateProfile';
+import React, { useEffect, useState } from 'react';
+import { useUpdateProfile } from '../../../useUpdateProfile';
 import { StyledHelpModalSelectOption } from '../ParametresHelpCard.styles';
 import { PARAMETRES_HELP_CARD_CONTENTS } from '../ParametresHelpCard.utils';
+import { UserProfile } from 'src/api/types';
 import { useModalContext } from 'src/components/modals/Modal';
+import { ModalGeneric } from 'src/components/modals/Modal/ModalGeneric';
 import { Button } from 'src/components/utils';
 import { H6 } from 'src/components/utils/Headings';
 import { SelectList } from 'src/components/utils/Inputs/SelectList';
 import { UserRole } from 'src/constants/users';
 import { useAuthenticatedUser } from 'src/hooks/authentication/useAuthenticatedUser';
 import { StyledParametresHelpModalCTAContainer } from './StyledParametresHelpModal.styles';
 
-export const ParametresHelpModal = ({ role }: { role: UserRole }) => {
+export const ParametresHelpModal = ({
+  role,
+  title,
+}: {
+  role: UserRole;
+  title: string;
+}) => {
   const { onClose } = useModalContext();
   const user = useAuthenticatedUser();
   const { userProfile } = user;
 
-  const { helpField, updateUserProfile, setTempProfile } = useProfile(
-    user,
-    onClose
-  );
+  const { helpField, updateUserProfile, closeModal } = useUpdateProfile(user);
+
+  const [tempProfile, setTempProfile] = useState<Partial<UserProfile>>({});
```

Comment:
Ici, pour éviter le `!(helpField in tempProfile)`, est-ce que tu peux pas créer un type intermédiaire du style `Partial<UserProfile> & { helpField: { name: HelpValue }[] }` ?

Comme ça c'est plus lisible (on override pas le type `UserProfile`).